sys/net/application_layer/ng_zep/ng_zep.c

```diff
@@ -0,0 +1,1072 @@
+    }
+    else {
+        hash2 = djb2_hash(cpuid + (CPUID_ID_LEN / 2), CPUID_ID_LEN / 2);
+    }
+
+    dev->eui64.u32[0] = hash1;
+    dev->eui64.u32[1] = hash2;
+
+    dev->eui64.u8[7] &= 0xfe;   /* set to unicast */
+    dev->eui64.u8[7] |= 0x02;   /* set to locally administered */
+#else
+    dev->addr = NG_ZEP_DEFAULT_ADDR_SHORT;
+    dev->eui64 = NG_ZEP_DEFAULT_ADDR_LONG;
+#endif
+    DEBUG("zep: initialized radio parameters: chan: %" PRIu8 ", pan: 0x%04" PRIx16
+          "addr: 0x%04" PRIx16 ", eui64: %016" PRIx64 "\n", dev->chan,
+          dev->pan, byteorder_ltobs(dev->addr).u16,
+          byteorder_ltobll(dev->eui64).u64);
+
+#ifdef MODULE_NG_SIXLOWPAN
+    dev->proto = NG_NETTYPE_SIXLOWPAN;
+#else
+    dev->proto = NG_NETTYPE_UNDEF;
+#endif
+
+    dev->seq = genrand_uint32();
+    dev->src_port = src_port;
+    dev->dst.u64[0] = dst->u64[0];
+    dev->dst.u64[1] = dst->u64[1];
+    dev->dst_port = dst_port;
+    dev->version = 2;
+    dev->lqi_mode = 1;
+
+    _pid = thread_create(_rx_stack, NG_ZEP_STACK_SIZE, NG_ZEP_PRIO,
+                         CREATE_STACKTEST, _event_loop, dev, "zep_app");
+
+    DEBUG("zep: started thread with PID %" PRIkernel_pid "\n", _pid);
+
+    return _pid;
+}
+
+/* helper functions for options to avoid type pruning */
+static inline void _set_uint16_ptr(uint16_t *ptr, uint16_t val)
+{
+    *ptr = val;
+}
+
+static inline void _set_uint64_ptr(uint64_t *ptr, uint64_t val)
+{
+    *ptr = val;
+}
+
+static inline void _set_flag_ptr(ng_netconf_enable_t *enable,
+                                 uint16_t flag_field, uint16_t flag)
+{
+    if (flag_field & flag) {
+        *enable = NETCONF_ENABLE;
+    }
+    else {
+        *enable = NETCONF_DISABLE;
+    }
+}
+
+static inline uint16_t *_get_uint16_ptr(void *ptr)
+{
+    return ptr;
+}
+
+static inline uint64_t *_get_uint64_ptr(void *ptr)
+{
+    return ptr;
+}
+
+static int _send(ng_netdev_t *netdev, ng_pktsnip_t *pkt)
+{
+    ng_zep_t *dev = (ng_zep_t *)netdev;
+    ng_pktsnip_t *ptr, *new, *hdr;
+    ng_zep_hdr_t *zep;
+    size_t payload_len = ng_pkt_len(pkt->next), hdr_len, mhr_offset;
+    ng_netreg_entry_t *entry;
+    uint8_t mhr[NG_IEEE802154_MAX_HDR_LEN], *data;
+    uint16_t fcs = 0;
+
+    if ((netdev == NULL) || (netdev->driver != &_zep_driver)) {
+        DEBUG("zep: wrong device on sending\n");
+        ng_pktbuf_release(pkt);
+        return -ENODEV;
+    }
+
+    /* create 802.15.4 header */
+    hdr_len = _make_data_frame_hdr(dev, mhr, (ng_netif_hdr_t *)pkt->data);
+
+    if (hdr_len == 0) {
+        DEBUG("zep: error on frame creation\n");
+        ng_pktbuf_release(pkt);
+        return -ENOMSG;
+    }
+
+    new = _zep_hdr_build(dev, hdr_len + payload_len + NG_IEEE802154_FCS_LEN, false);
+
+    if (new == NULL) {
+        DEBUG("zep: could not allocate ZEP header in pktbuf\n");
+        ng_pktbuf_release(pkt);
+        return -ENOBUFS;
+    }
+
+    zep = new->data;
```

Comment:
Not a big issue, but wouldn't it be easier readable if you make this assignment just before you need it, i.e. at line 261?